### PR TITLE
Add godoc example

### PR DESCRIPTION
**Pull request by embano1**
Add a `go test`-able example which also shows up in godoc and https://pkg.go.dev/ and also allows user to run ("interact") with the code.

Closes: #70
Signed-off-by: Michael Gasch <mgasch@vmware.com>

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	if err != nil {
+		log.Fatalf("could not create quamina instance: %v", err)
+	}
+
+	err = q.AddPattern("premium user", premiumUserPattern)
+	if err != nil {
+		log.Fatalf("could not add pattern: %v", err)
+	}
+
+	event, err := q.MatchesForEvent([]byte(userRegisteredEvent))
```
Comment (timbray) [inline example_test.go:36]: not crazy about the variable name "event" - what you get back is []X, so a better name would be "matches" or something like that.

File: example_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	if err != nil {
+		log.Fatalf("could not match for event: %v", err)
+	}
+
+	if event != nil {
```
Comment (timbray) [inline example_test.go:41]: I'm pretty sure that Quamina never returns nil. It always returns []X which of course might be empty.  The kind of check you want to do is something like 
```go
if len(matches) == 0 || matches[0] != "premium user" {
  t.Error(whatever)
}
```

**Review by timbray — CHANGES_REQUESTED** [review]
This is useful because it reveals that that the existing documentation did not properly explain the return value from `MatchesForEvent`.

File: example_test.go (hunk 1 above)
Comment (embano1) [reply example_test.go:36]: Ah, of course! Will do.

File: example_test.go (hunk 2 above)
Comment (embano1) [reply example_test.go:41]: Ah! It's even in the docs. I just followed go "idioms" here, but you're right: it preallocates a slice!

Comment (timbray) [reply example_test.go:41]: Yes, that makes the calling sequence a little simpler for the caller, no need to check for nil.

File: example_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	for _, m := range matches {
+		if m == patternName {
+			fmt.Printf("pattern matched for event: %q", patternName)
+			return
+		}
+	}
+
+	fmt.Println("no pattern match")
```
Comment (embano1) [inline example_test.go:49]: @timbray I think this is even better/more clear now

File: example_test.go (hunk 2 above)
Comment (embano1) [reply example_test.go]: I slightly improved it for better readability/to follow along the example.

**Comment by timbray** [comment]
That's weird, slight decrease in coverage?  That doesn't make any sense to me, does it to you?

**Comment by embano1** [comment]
> That's weird, slight decrease in coverage?  That doesn't make any sense to me, does it to you?

No, no idea...

**Pull request merged**